Number_types/include/CGAL/Quotient.h

```diff
@@ -686,11 +707,547 @@ template < class NT > class Real_embeddable_traits_quotient_base< Quotient<NT> >
     class To_interval
       : public CGAL::cpp98::unary_function< Type, std::pair< double, double > > {
       public:
+
+        // =========================================================================== //
+
+        // CURRENT VERSION
+
+        // =========================================================================== //
+
+        std::pair< std::pair<double, double>, long > get_interval_exp( NT& x ) const {
+
+          CGAL_assertion(x >= 0);
+          const unsigned n = msb(x);
+          const unsigned num_dbl_digits = std::numeric_limits<double>::digits;
+          if (n > num_dbl_digits) {
+
+            const unsigned d = n - num_dbl_digits;
+            x = x >> d;
+
+            const double l = boost::multiprecision::detail::do_cast<double>(x);
+            const double u = boost::multiprecision::detail::do_cast<double>(x + 1);
+            return std::make_pair( std::make_pair(l, u), d );
+
+          } else {
+
+            const double l = boost::multiprecision::detail::do_cast<double>(x);
+            const double u = l;
+            return std::make_pair( std::make_pair(l, u), 0 );
+          }
+        }
+
+        std::pair<double, double> get_interval_as_gmpzf( Type& x ) const {
+
+          double l = 0.0, u = 0.0;
+          if (x.num == 0) { // return [0.0, 0.0]
+            return std::make_pair(l, u);
+          }
+          CGAL_assertion(x.num != 0);
+          CGAL_assertion(x.den != 0);
+
+          // Handle signs.
+          bool change_sign = false;
+          if (x.num < 0 && x.den < 0) {
+            x.num = -x.num;
+            x.den = -x.den;
+          } else if (x.num < 0 && x.den > 0) {
+            change_sign = true;
+            x.num = -x.num;
+          } else if (x.num > 0 && x.den < 0) {
+            change_sign = true;
+            x.den = -x.den;
+          }
+          CGAL_assertion(x.num > 0 && x.den > 0);
+
+          Protect_FPU_rounding<true> P(CGAL_FE_TONEAREST);
```

Comment:
I think it does not affect the results in this case but I left it because we figured out that when converting cpp_int to double with boost using `x.convert_to<double>()` we need to set the rounding mode to nearest. I think it can be removed here. I will check.

Reply:
I have removed that.